src/modules/sway/hide.cpp

```diff
@@ -0,0 +1,45 @@
+#include <iostream>
+#include "modules/sway/hide.hpp"
+#include <spdlog/spdlog.h>
+#include "client.hpp"
+#include "wlr-layer-shell-unstable-v1-client-protocol.h"
+
+namespace waybar::modules::sway {
+
+Hide::Hide(const std::string& id, const Bar& bar, const Json::Value& config)
+    : ALabel(config, "hide", id, "{}", 0, true), bar_(bar), windowId_(-1) {
+  ipc_.subscribe(R"(["bar_state_update","barconfig_update"])");
+  ipc_.signal_event.connect(sigc::mem_fun(*this, &Hide::onEvent));
+  bar_.window.get_style_context()->add_class("hidden");
+  // Launch worker
+  worker();
+}
+
+void Hide::onEvent(const struct Ipc::ipc_response& res) {
+  auto payload = parser_.parse(res.payload);
+  if (payload.isMember("mode")) {
```

Comment:
Note that each waybar instance supposed to have bar id (`waybar -b bar-0`) and events from sway could be addressed to the specific bar or to all bars.
Somewhere around here you should check if the current bar needs to handle the event.

Reply:
I was going to use just call `setVisible` on the  `bar_` object, but I got an error about mismatching `const` which is above my meagre C++-fu.